src/copilot/inspect/InspectionCopilot.ts

```diff
@@ -0,0 +1,256 @@
+        public EmployeePojo(String name) {
+            this.name = name;
+        }
+        public String getRole() {
+            String result = '';
+            // @PROBLEM: Using if-else statements to check the type of animal
+            // @SOLUTION: Use switch expression
+            // @SYMBOL: if
+            // @SEVERITY: MIDDLE
+            if (this.name.equals("Miller")) {
+                result = "Senior";
+            } else if (this.name.equals("Mike")) {
+                result = "HR";
+            } else {
+                result = "FTE";
+            }
+            return result;
+        }
+        public void test(String[] arr) {
+            try {
+                Integer.parseInt(arr[0]);
+            } catch (Exception e) {
+                // @PROBLEM: Print stack trace in case of an exception
+                // @SOLUTION: Log errors to a logger
+                // @SYMBOL: ex.printStackTrace
+                // @SEVERITY: LOW
+                e.printStackTrace();
+            }
+        }
+    }
+    //${Copilot.DEFAULT_END_MARK}
+    `;
+
+    // Initialize regex patterns
+    private static readonly PROBLEM_PATTERN: RegExp = /\/\/ @PROBLEM: (.*)/;
+    private static readonly SOLUTION_PATTERN: RegExp = /\/\/ @SOLUTION: (.*)/;
+    private static readonly SYMBOL_PATTERN: RegExp = /\/\/ @SYMBOL: (.*)/;
+    private static readonly LEVEL_PATTERN: RegExp = /\/\/ @SEVERITY: (.*)/;
+
+    private readonly debounceMap = new Map<string, NodeJS.Timeout>();
+
+    public constructor() {
+        const messages: { role: string, content: string }[] = [
+            { role: "system", content: InspectionCopilot.SYSTEM_MESSAGE },
+            { role: "user", content: InspectionCopilot.EXAMPLE_USER_MESSAGE },
+            { role: "assistant", content: InspectionCopilot.EXAMPLE_ASSISTANT_MESSAGE },
+        ];
+        super(messages);
+    }
+
+    /**
+     * inspect the given code using copilot and return the inspections
+     * @param code code to inspect
+     * @param key key to debounce the inspecting, if provided
+     * @param wait debounce time in milliseconds, default is 3000ms
+     * @returns inspections provided by copilot
+     */
+    public inspectCode(code: string, key?: string, wait?: number): Promise<Inspection[]> {
+        const _doInspectCode: (code: string) => Promise<Inspection[]> = instrumentSimpleOperation("java.copilot.inspect.code", this.doInspectCode.bind(this));
+        if (!key) { // inspect code immediately without debounce
+            return _doInspectCode(code);
+        }
+        // inspect code with debounce if key is provided
+        if (this.debounceMap.has(key)) {
+            clearTimeout(this.debounceMap.get(key) as NodeJS.Timeout);
+            logger.debug(`debounced`, key);
+        }
+        return new Promise<Inspection[]>((resolve) => {
+            this.debounceMap.set(key, setTimeout(() => {
+                void _doInspectCode(code).then(inspections => {
```

Comment:
what's the purpose to add a prefix `void`?

Reply:
to prevent eslint warning `unused return value`